pithos/pithos.py

```diff
@@ -161,18 +161,23 @@ def __init__(self, app, test_mode):
         self.set_proxy(reconnect=False)
         self.set_audio_quality()
 
-        email = self.settings['email']
         try:
-            password = get_account_password(email)
+            unlock_keyring()
         except GLib.Error as e:
             if e.code == 2:
-                self.fatal_error_dialog(e.message, _('You need to install a service such as gnome-keyring.'))
+                logging.error('You need to install a service such as gnome-keyring. Error: {}'.format(e))
+                self.fatal_error_dialog(
+                    e.message,
+                    _('You need to install a service such as gnome-keyring.'),
+                )
+
+        email = self.settings['email']
+        password = get_account_password(email)
```

Comment:
> Once we get past the unlocking of the keyring I don't think anything else should raise any exceptions.

With any external resource you have to remember that your running a multi-process system, between two calls in your program the keyring can change state. This is uncommon of course but there is no reason to not handle edge cases.